Xamarin.Forms.Core/IndicatorsView.cs

```diff
@@ -0,0 +1,285 @@
+		}
+
+		public DataTemplate IndicatorTemplate
+		{
+			get => GetValue(IndicatorTemplateProperty) as DataTemplate;
+			set => SetValue(IndicatorTemplateProperty, value);
+		}
+
+		public IndicatorShape IndicatorShape
+		{
+			get => (IndicatorShape)GetValue(IndicatorShapeProperty);
+			set => SetValue(IndicatorShapeProperty, value);
+		}
+
+		public Color IndicatorsColor
+		{
+			get => (Color)GetValue(IndicatorsColorProperty);
+			set => SetValue(IndicatorsColorProperty, value);
+		}
+
+		public Color SelectedIndicatorColor
+		{
+			get => (Color)GetValue(SelectedIndicatorColorProperty);
+			set => SetValue(SelectedIndicatorColorProperty, value);
+		}
+
+		public double IndicatorsSize
+		{
+			get => (double)GetValue(IndicatorsSizeProperty);
+			set => SetValue(IndicatorsSizeProperty, value);
+		}
+
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public IEnumerable ItemsSource
+		{
+			get => GetValue(ItemsSourceProperty) as IEnumerable;
+			set => SetValue(ItemsSourceProperty, value);
+		}
+
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public new IList<View> Children => base.Children;
+
+		public static void SetItemsSourceBy(IndicatorsView indicatorsView, CarouselView carouselView)
+			=> indicatorsView.SetItemsSourceBy(carouselView);
+
+		public void SetItemsSourceBy(CarouselView carouselView)
+		{
+			SetBinding(PositionProperty, new Binding
+			{
+				Path = nameof(CarouselView.Position),
+				Source = carouselView
+			});
+			SetBinding(ItemsSourceProperty, new Binding
+			{
+				Path = nameof(ItemsView.ItemsSource),
+				Source = carouselView
+			});
+		}
+
+		protected virtual void ApplySelectedStyle(View view, int index)
+			=> view.BackgroundColor = SelectedIndicatorColor;
+
+		protected virtual void ApplyUnselectedStyle(View view, int index)
+			=> view.BackgroundColor = IndicatorsColor;
+
+		private void AddExtraIndicatorsItems()
+		{
+			var oldCount = Children.Count;
+			for (var i = 0; i < Count - oldCount && i < MaxVisibleCount - oldCount; ++i)
+			{
+				var indicator = IndicatorTemplate?.CreateContent() as View ?? new Frame
+				{
+					Padding = 0,
+					HasShadow = false,
+					BorderColor = Color.Transparent,
+					VerticalOptions = LayoutOptions.Center,
+					HorizontalOptions = LayoutOptions.Center,
+					WidthRequest = IndicatorsSize,
+					HeightRequest = IndicatorsSize,
+					CornerRadius = IndicatorShape == IndicatorShape.Circle
+						? (float)IndicatorsSize / 2
+						: 0
+				};
+				var itemTapGesture = new TapGestureRecognizer();
+				itemTapGesture.Tapped += (tapSender, tapArgs) => Position = IndexOf(tapSender as View);
+				indicator.GestureRecognizers.Add(itemTapGesture);
+				Children.Add(indicator);
+			}
+		}
+
+		private void RemoveRedundantIndicatorsItems()
+		{
+			foreach (var item in Children.Where((v, i) => i >= Count).ToArray())
+			{
+				Children.Remove(item);
+			}
+		}
+
+		private int IndexOf(View view) => Children.IndexOf(view);
+
+		private void ApplyStyle(View view)
```

Comment:
the Style name is unfortunate here

Reply:
sure
I had an idea, that uses coud extend and override these methods for customizing "style". Not back color only

Reply:
Also drop the above private keywords. There's a lot more throughout the code.

Reply:
@adrianknight89 nice note, sure